libs/ktxreader/include/ktxreader/Ktx2Reader.h

```diff
@@ -85,16 +93,107 @@ class Ktx2Reader {
          */
         Texture* load(const uint8_t* data, size_t size, TransferFunction transfer);
 
+        /**
+         * Asynchronous Interface
+         * ======================
+         *
+         * Alternative API suitable for asynchronous transcoding of mipmap levels.
+         * If unsure that you need to use this, then don't, just call load() instead.
+         * Usage pseudocode:
+         *
+         *    auto async = reader->asyncCreate(data, size, TransferFunction::LINEAR);
+         *    mTexture = async->getTexture();
+         *    auto backgroundThread = spawnThread({ async->doTranscoding(); })
+         *    backgroundThread.wait();
+         *    async->uploadImages();
+         *    reader->asyncDestroy(async);
+         *
+         * In the documentation comments, "foreground thread" refers to the thread that the
+         * Filament Engine was created on.
+         */
+        class Async {
+        public:
+            /**
+             * Retrieves the Texture object.
+             *
+             * The texture is available immediately, but does not have its miplevels ready until
+             * after uploadImages(). The caller has ownership over this texture and is responsible
+             * for freeing it.
+             */
+            Texture* getTexture() const noexcept;
+
+            /**
+             * Loads mipmaps from the KTX2 file and transcodes them to the resolved format.
+             *
+             * This is typically called from a background thread.
+             */
+            Result doTranscoding();
+
+            /**
+             * Uploads all mipmaps to the texture.
+             *
+             * This calls Texture::setImage() and potentially generates missing mipmaps.
+             * If transcoding has not yet completed, returns INCOMPLETE.
+             *
+             * NOTE: This can be called only from the foreground thread.
+             */
+            void uploadImages();
```

Comment:
What happens if it's not called from the right thread? 

It says it returns `INCOMPLETE` in some cases, but it doesn't have a return value.